### PR TITLE
docs(cli): added yarn global bin instruction to CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes
When setting up environment from scratch (brand new mac with zero tools installed) I noticed that we have the following gaps:
- DynamoDB simulator depends on `java` which is not explicitly mentioned in setup guide.
-  it's hard to discover where `amplify-dev` command is. Therefore adding a line in the CONTRIBUTING guide to add Yarn's global bin to PATH - that's where the `amplify-dev` ends up being.

The result can be seen [here](https://github.com/sobolk/amplify-cli/blob/add-yarn-global-to-contribution-guide/CONTRIBUTING.md#local-environment-setup)

#### Checklist

- [ x] PR description included

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ This section should get you running with **Amplify CLI** and get you familiar wi
 
    > If you are using Yarn v2, run `yarn set version classic` to change to Yarn Classic.
 
+   > Ensure that [Yarn global bin](https://classic.yarnpkg.com/en/docs/cli/global) is added to your PATH. For example, add `export PATH="$(yarn global bin):$PATH"` to your shell profile file on Linux or macOS.
+
+1. Ensure you have [Java](https://adoptopenjdk.net/) installed and `java` command is available in your system. This is required for DynamoDB emulator.
```
Comment (jhockett) [inline CONTRIBUTING.md:42]: Small nit, can you bump the numbers that follow by 1?

**Review by jhockett — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (jhockett) [reply CONTRIBUTING.md:42]: Also, that link looks out of date. I think the correct link would be https://adoptium.net/

Comment (sobolk) [reply CONTRIBUTING.md:42]: They're all `1.` see the [raw source](https://raw.githubusercontent.com/aws-amplify/amplify-cli/dev/CONTRIBUTING.md). Markdown auto numbers the points.

Comment (sobolk) [reply CONTRIBUTING.md]: Link fixed. Thanks for catching this.

**Review by jhockett — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,6 +37,10 @@ This section should get you running with **Amplify CLI** and get you familiar wi
 
    > If you are using Yarn v2, run `yarn set version classic` to change to Yarn Classic.
 
+   > Ensure that [Yarn global bin](https://classic.yarnpkg.com/en/docs/cli/global) is added to your PATH. For example, add `export PATH="$(yarn global bin):$PATH"` to your shell profile file on Linux or macOS.
+
+1. Ensure you have [Java](https://adoptium.net/) installed and `java` command is available in your system. This is required for DynamoDB emulator.
```
Comment (edwardfoyle) [inline CONTRIBUTING.md:42]: Might be better to point people here: https://aws.amazon.com/corretto/ to download java

**Review by edwardfoyle — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (jhockett) [reply CONTRIBUTING.md:42]: Sorry @sobolk, I should have pointed you here ^

**Pull request merged**